### PR TITLE
improve(TransactionUtils): Use sdk-v2 gasPriceOracle

### PR DESCRIPTION
**Pull request by pxrl**
The gasPriceOracle was recently upgraded to support Polygon's gas station.

As a nice side-effect, we can remove node-fetch from our list of dependencies.

Fixes ACX-1434.

### CODE REVIEW

File: src/utils/TransactionUtils.ts [hunk 1]
```diff
@@ -134,23 +134,23 @@ export async function getGasPrice(
...
-    return {
-      maxFeePerGas: scaleByNumber(feeData.maxFeePerGas, priorityScaler * maxFeePerGasScaler), // scale up the maxFeePerGas. Any extra paid on this is refunded.
-      maxPriorityFeePerGas: scaleByNumber(feeData.maxPriorityFeePerGas, priorityScaler),
-    };
-  } else {
-    return { gasPrice: scaleByNumber(feeData.gasPrice, priorityScaler) };
+  const chainId = (await provider.getNetwork()).chainId;
+  const feeData = await gasPriceOracle.getGasPriceEstimate(provider, chainId);
+
+  if (feeData.maxPriorityFeePerGas.gt(feeData.maxFeePerGas)) {
```
Comment (nicholaspai) [inline src/utils/TransactionUtils.ts:140]: Is there always guaranteed to be a `maxPriorityFee` and `maxFeePerGas` even in legacy networks?

**Review by nicholaspai — APPROVED** [review]

File: src/utils/TransactionUtils.ts (hunk 1 above)
Comment (pxrl) [reply src/utils/TransactionUtils.ts:140]: No, but it doesn't actually matter. The output of `getGasPriceEstimate` has already been normalised into the  `GasPriceEstimate` format: `{ maxFeePerGas: BigNumber, maxPriorityFeePerGas: BigNumber }`. So for legacy networks, `maxPriorityFeePerGas` is 0.

Comment (james-a-morris) [reply src/utils/TransactionUtils.ts:140]: +1 for the normalization & abstraction @pxrl

Comment (nicholaspai) [reply src/utils/TransactionUtils.ts:140]: nice

File: src/utils/TransactionUtils.ts [hunk 2]
```diff
@@ -134,23 +134,23 @@ export async function getGasPrice(
...
-    if (feeData.maxPriorityFeePerGas.gt(feeData.maxFeePerGas)) {
-      feeData.maxFeePerGas = scaleByNumber(feeData.maxPriorityFeePerGas, 1.5);
-    }
-    return {
-      maxFeePerGas: scaleByNumber(feeData.maxFeePerGas, priorityScaler * maxFeePerGasScaler), // scale up the maxFeePerGas. Any extra paid on this is refunded.
-      maxPriorityFeePerGas: scaleByNumber(feeData.maxPriorityFeePerGas, priorityScaler),
-    };
-  } else {
-    return { gasPrice: scaleByNumber(feeData.gasPrice, priorityScaler) };
+  const chainId = (await provider.getNetwork()).chainId;
```
Comment (james-a-morris) [inline src/utils/TransactionUtils.ts:137]:
```suggestion
  const { chainId}  = (await provider.getNetwork());
```
Nit:

**Review by james-a-morris — APPROVED** [review]
LGTM

**Pull request merged**